src/hooks/useAuth.ts

```
@@ -7,6 +7,36 @@ import { CleverAuthPlugin } from "../auth/cleverAuthPlugin";
 
 const SAML_AUTH_TYPE = "http://librarysimplified.org/authtype/SAML-2.0";
 
+const TOKEN_NOT_FOUND = "TOKEN_NOT_FOUND";
+
+export function getTokenFromUrl(
+  router
+): {
+  token:
+    | string
+    | { credentials?: string; undefined; error?: string | undefined }
+    | undefined;
+  type: string;
+} {
+  const cleverAccessToken =
+    typeof window !== "undefined" && CleverAuthPlugin.lookForCredentials();
+
+  const { access_token: samlAccessToken } = router.query;
+
+  let type = TOKEN_NOT_FOUND;
+  /* there should never be a case where both types of access tokens exists at the same time */
```

For another PR, maybe throw an error if this case happens.